pkg/component/gardenerapiserver/gardener_apiserver_test.go

```diff
@@ -177,50 +177,71 @@ var _ = Describe("GardenerAPIServer", func() {
-				Selector: &metav1.LabelSelector{MatchLabels: map[string]string{
-					"app":  "gardener",
-					"role": "apiserver",
-				}},
-			},
+		podDisruptionBudgetFor = func(k8sGreaterEqual126 bool) *policyv1.PodDisruptionBudget {
+			var (
+				unhealthyPodEvictionPolicyAlwatysAllow = policyv1.AlwaysAllow
+				pdb                                    = &policyv1.PodDisruptionBudget{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "gardener-apiserver",
+						Namespace: namespace,
+						Labels: map[string]string{
+							"app":  "gardener",
+							"role": "apiserver",
+						},
+					},
+					Spec: policyv1.PodDisruptionBudgetSpec{
+						MaxUnavailable: gardenerutils.IntStrPtrFromInt32(1),
+						Selector: &metav1.LabelSelector{MatchLabels: map[string]string{
+							"app":  "gardener",
+							"role": "apiserver",
+						}},
+					},
+				}
+			)
+
+			if k8sGreaterEqual126 {
+				pdb.Spec.UnhealthyPodEvictionPolicy = &unhealthyPodEvictionPolicyAlwatysAllow
+			}
+
+			return pdb
 		}
-		serviceRuntime = &corev1.Service{
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      "gardener-apiserver",
-				Namespace: namespace,
-				Annotations: map[string]string{
-					"networking.resources.gardener.cloud/from-all-webhook-targets-allowed-ports": `[{"protocol":"TCP","port":8443}]`,
-					"service.kubernetes.io/topology-mode":                                        "auto",
-				},
-				Labels: map[string]string{
-					"app":  "gardener",
-					"role": "apiserver",
-					"endpoint-slice-hints.resources.gardener.cloud/consider": "true",
+		serviceRuntimeFor = func(k8sGreaterEqual127 bool) *corev1.Service {
+			svc := &corev1.Service{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "gardener-apiserver",
+					Namespace: namespace,
+					Annotations: map[string]string{
+						"networking.resources.gardener.cloud/from-all-webhook-targets-allowed-ports": `[{"protocol":"TCP","port":8443}]`,
+					},
+					Labels: map[string]string{
+						"app":  "gardener",
+						"role": "apiserver",
+						"endpoint-slice-hints.resources.gardener.cloud/consider": "true",
+					},
 				},
-			},
-			Spec: corev1.ServiceSpec{
-				Type: corev1.ServiceTypeClusterIP,
-				Selector: map[string]string{
-					"app":  "gardener",
-					"role": "apiserver",
+				Spec: corev1.ServiceSpec{
+					Type: corev1.ServiceTypeClusterIP,
+					Selector: map[string]string{
+						"app":  "gardener",
+						"role": "apiserver",
+					},
+					Ports: []corev1.ServicePort{{
+						Port:       443,
+						Protocol:   corev1.ProtocolTCP,
+						TargetPort: intstr.FromInt32(8443),
+					}},
 				},
-				Ports: []corev1.ServicePort{{
-					Port:       443,
-					Protocol:   corev1.ProtocolTCP,
-					TargetPort: intstr.FromInt32(8443),
-				}},
-			},
+			}
+
+			if k8sGreaterEqual127 {
+				svc.Annotations["service.kubernetes.io/topology-mode"] = "auto"
+			} else {
+				svc.Annotations["service.kubernetes.io/topology-aware-hints"] = "auto"
+			}
```

Comment:
What is the reasoning for this change? It does not relate to the pod disruption budget story, right?